flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileMonitoringTest.java

```diff
@@ -106,6 +109,140 @@ public static void destroyHDFS() {
 	//						TESTS
 
 	@Test
+	public void testFileReadingOperatorWithIngestionTime() throws Exception {
+		Set<org.apache.hadoop.fs.Path> filesCreated = new HashSet<>();
+		Map<Integer, String> expectedFileContents = new HashMap<>();
+		for(int i = 0; i < NO_OF_FILES; i++) {
+			Tuple2<org.apache.hadoop.fs.Path, String> file = fillWithData(hdfsURI, "file", i, "This is test line.");
+			filesCreated.add(file.f0);
+			expectedFileContents.put(i, file.f1);
+		}
+
+		TextInputFormat format = new TextInputFormat(new Path(hdfsURI));
+		TypeInformation<String> typeInfo = TypeExtractor.getInputFormatTypes(format);
+
+		ContinuousFileReaderOperator<String, ?> reader = new ContinuousFileReaderOperator<>(format);
+
+		StreamConfig streamConfig = new StreamConfig(new Configuration());
+		streamConfig.setTimeCharacteristic(TimeCharacteristic.IngestionTime);
+
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		executionConfig.setAutoWatermarkInterval(100);
+
+		TestTimeServiceProvider timeServiceProvider = new TestTimeServiceProvider();
+		OneInputStreamOperatorTestHarness<FileInputSplit, String> tester =
+			new OneInputStreamOperatorTestHarness<>(reader, executionConfig, timeServiceProvider, streamConfig);
+
+		reader.setOutputType(typeInfo, new ExecutionConfig());
+		tester.open();
+
+		timeServiceProvider.setCurrentTime(0);
+
+		long elementTimestamp = 201;
+		timeServiceProvider.setCurrentTime(elementTimestamp);
+
+		// test that a watermark is actually emitted
+		Assert.assertTrue(tester.getOutput().size() == 1 &&
+			tester.getOutput().peek() instanceof Watermark &&
+			((Watermark) tester.getOutput().peek()).getTimestamp() == 200);
```

Comment:
You don't need to change it, but I think it's a good idea to test the conditions independently. This allows you to see which condition was false, based on the line number.